### ISSUE TITLE
[GiD] Cut post-process file to reduce the size

### ISSUE THREAD

**Issue by riccardotosi** [KratosMultiphysics/Kratos#6844]
Dear all,

I have an enormous GiD post-process  "*bin" file, whose size is around 80 gigabytes, stored in a supercomputer, and I would like to post-process it.
The file solves a FluidDynamics application problem, then it has many time steps. The initial time steps are of no physical interest, so I would like to discard them to make the file lighter. These initial time steps are, more or less, 15% of the simulation length.

Is there a way to reduce the file considering only some time steps, and not all of them? I know it is possible to stop the copy (from the supercomputer to my machine) and this way we would have a lighter *bin file, but doing so I will have only the initial time steps, which have no physical interest.

On the other hand, I would like to choose which time steps I want.

Thanks.

**Comment by rubenzorrilla** [comment]
I think that what you need can be achieved by changing the `SingleFile` flag in the GiD output process settings to `MultipleFiles`. By doing this you will get a single (or as many as MPI cores if you are running in parallel) file for each time step.

Then you can discard the ones you do not require any more.

**Comment by jginternational** [comment]
I understand that you expect to 'cut it' in the current 80Gb file, instead of calculating again

Maybe @pasenau can help

**Comment by riccardotosi** [comment]
Thanks @rubenzorrilla , I didn't know about this and I will change for the future. Yes, I would prefer to cut the GiD file I already have, if possible. For example, I would like to cut from time = 100 seconds to time = 120 seconds.

**Comment by philbucher** [comment]
you can also reduce the frequencs with which the output is written
I suggest to check the available settings to see what can be done

**Comment by riccardotosi** [comment]
Hi @philbucher . Yes I knew about this and I am already using this feature. However, I would avoid having a too big frequency, since I think may be not optimal for post-processing.

**Comment by pasenau** [comment]
Hi @riccardotosi , 

in GiD you can use the **Result cache** option in the _preferences window --> Postprocess --> File_. When this option is enabled, GiD will only load the result you're visualizing, after first creating an index file once. You can also set the amount of Memory you want to use to hold the results in memory.

At the moment there is no easy way to delete the unwnated steps. 
(There is an option to convert the binary file to hdf5 (which gid also understands) and then delete the unwanted steps with an external tool), 
well two, but one of them is using command line flags but does not work in 14.1.8d (i just tested it), and the other one is using commands:

`./gidx -n -e Postprocess Mescape Files Read YourInputFile.pot.bin Mescape Files Saveall HDF5GidPost YourOutputFile.post.h5`

(the _-n_ option is to use gid without gui, and the _-e_ option is to provide commands to gid, like the command line in the graphical window)

GiD can read this hdf5 file too and the Result cache will access directly this hdf5 file, without any index file.

Well there is a third more cumbersome option: convert to ascii, split the file, edit some chunks and glue them together.....

In the future we'll add an option to delete the steps you don't want in gid.

![PreferencesPostResultsCache](https://user-images.githubusercontent.com/20422435/80709508-af3b4200-8aed-11ea-9cfc-181dc3430d5f.png)

**Comment by riccardotosi** [comment]
Thanks @pasenau , I will try what you suggested!
